source/common/secret/sds_api.cc

```diff
@@ -15,13 +15,21 @@ namespace Secret {
 SdsApi::SdsApi(const LocalInfo::LocalInfo& local_info, Event::Dispatcher& dispatcher,
                Runtime::RandomGenerator& random, Stats::Store& stats,
                Upstream::ClusterManager& cluster_manager, Init::Manager& init_manager,
-               const envoy::api::v2::core::ConfigSource& sds_config, std::string sds_config_name)
+               const envoy::api::v2::core::ConfigSource& sds_config, std::string sds_config_name,
+               Secret::SecretManager& secret_manager)
     : local_info_(local_info), dispatcher_(dispatcher), random_(random), stats_(stats),
       cluster_manager_(cluster_manager), sds_config_(sds_config), sds_config_name_(sds_config_name),
-      secret_hash_(0) {
+      secret_hash_(0), id_(std::to_string(MessageUtil::hash(sds_config)) + sds_config_name),
```

Comment:
can we pass in id from the constructor so it is always the same as calculated by sercret_manager.